docs/whichkey/reference.md

```diff
@@ -0,0 +1,281 @@
+
+### Repeat Recent
+
+Command: `whichkey.repeatRecent`
+
+Description:
+A command to show a which-key menu with key 1 to 9 to repeat the most recently executed command bindings.
+
+### Recent Most Recent
+
+Command: `whichkey.repeatMostRecent`
+
+Description:
+A command to repeat the command binding most recently executed on which-key.
+
+### Toggle Zen Mode
+
+Command: `whichkey.toggleZenMode`
+
+Description:
+A command to toggle zen mode for transient menu, which will show/hide all the menu items to save screen real estate.
+
+### Open File
+
+Command: `whichkey.openFile`
+
+Description:
+This is a command to get around issue where vscode doesn't provide a single cross platform command to open file.
+See [vscode-which-key#26](https://github.com/VSpaceCode/vscode-which-key/issues/26).
+
+## Context
+
+The extension will also set the following context when applicable so it can be used in `when` clause of vscode shortcuts.
+
+`whichkeyActive` is a boolean that will be set to `true` when whichkey is active including the time when it is excluding a command.
+This is rarely used and most of the use case can be replaced with `whichKeyVisible`
+
+`whichkeyVisible` is a boolean that will be set to `true` when whichkey menu is visible. Note that this will be `false` when transient menu is visible.
+
+`transientVisible` is a boolean that wil lbe set to `true` when the transient menu is visible.
+
+## Config
+
+The extension uses the following configurations.
+
+### Delay
+
+Key: `whichkey.delay`
+
+Type: `number`
+
+Description:
+Delay (in milliseconds) for which-key menu to display. Setting this to positive value will delay the display of the which-key menu items while
+key input remains functional.
+
+### Show Icons
+
+Key: `whichkey.showIcons`
+
+Type: `boolean`
+
+Description:
+Controls whether to show or hide icons in which-key menu.
+
+### Sort Order
+
+Key: `whichkey.sortOrder`
+
+Type: `"none"` | `"alphabetically"` | `"nonNumberFirst"`
+
+Description:
+Controls the sorting order of the which-key menu items.
+
+- `"none"` will not sort the bindings.
+- `"alphabetically"` will sort the bindings alphabetically using [`String.prototype.localeCompare()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare).
+- `"nonNumberFirst"` will sort the bindings alphabetically like `"alphabetically"` but have the bindings with number as key last.
+
+### Which-key Default Bindings
+
+Key: `whichkey.bindings`
+
+Type: `Array<BindingItem>`
+
+Description:
+The configuration of the default which-key menu. This contains the array of [`BindingItem`](#bindingitem).
+
+### Transient Bindings
+
+Key: `whichkey.transient`
+
+Type: `object`
+
+Description:
+A key-value pair to store default definitions of transient menu for command `whichkey.showTransient` to reference.
+
+## Shortcut
+
+These are the shortcut the extension contributed, which means they will work without manually editing your user's `keybindings.json`.
```

Comment:
```suggestion
These are the shortcuts the extension contributed, which means they will work without manually editing your user's `keybindings.json`.
```

Reply:
I don't understand `the extension contributed`.
Is it better `boundled with the extension`?

Reply:
bundled is probably a term easier to understand. vscode use contribute to for extension to "contribute" to settings, shortcuts, lsp, etc.

VScode also uses contribution in the extension UI.
<img width="587" alt="image" src="https://user-images.githubusercontent.com/3455662/121463457-ef165f00-c966-11eb-82ab-9730893bc4e6.png">

```suggestion
The whichkey extension contributes the following shortcuts, which means they will work without manually editing your user's `keybindings.json`.
```